src/overlays/actors/ovl_En_Mu/z_en_mu.h

```diff
@@ -15,8 +15,8 @@ typedef struct EnMu {
     /* 0x0194 */ ColliderCylinder collider;
     /* 0x01E0 */ NpcInteractInfo npcInfo;
     /* 0x0208 */ u16 defaultTextId;
-    /* 0x020A */ s16 unk_20A[16];
-    /* 0x022A */ s16 unk_22A[17];
+    /* 0x020A */ s16 fidgetTableY[16];
+    /* 0x022A */ s16 fidgetTableZ[17]; // @note: looks like a bug, usually tables match
```

Comment:
```suggestion
    /* 0x022A */ s16 fidgetTableZ[16];
```

This should match.

Reply:
i'm a bit cautious to change something substantially enough, like order of variables or parameters, even if it should be commutative in regards to what functions return: in my perception of "matching recompilation" asm instructions - and then machine bytes - could be reordered by this actions.

if compiled data should match perfectly, we should match structs sizes too. probably a better option would be
```
typedef struct EnMu {
    ...
    /* 0x022A */ s16 fidgetTableZ[16];
    /* 0x024A */ s16 unk_24A;
} EnMu; // size = 0x024C
```
additionally, even if an "OoB" read happens, it will address `unk_24A`

UPD 211263295cd8768d63c1426025cf913ba6bedc18

Reply:
It is not the style of the project to add fake variables when not necessary for matching. If you can remove the variable, compile across all versions and it still matches, then it shouldn't exist. Here, because the variable located within a struct, you can get away with just compiling one version to verify `unk_24A` is fake.

Furthermore, adding a dummy variable after the array would not be the appropriate way to handle an OoB array read, should one exist (and a quick code review reveals that no OoB read should occur in normal situations). In this case it's particularly obvious that it fixes nothing since `fidgetTableY` and `fidgetTableZ` use the same index value; a read/write into index =16 would end up corrupting `fidgetTableZ[0]` on IDO.

Reply:
understood. so, as a conclusion:
- look up `AVOID_UB` (i see there's a handful of applications to base off)
- for future reference, think in the said key: fix, test, add optional case

it's all new to new, no surprise. thank you for explanations

UPD e2578f456e239b8a84e59d10fe8911b646fbad9a: turns out i didn't spot the automatic padding of exactly 2 bytes; in my defense, i was sure the sizes decresed without it, according to my tooling. eh, nevermind

Reply:
To be clear, we arent fixing an issue with the original code here. We are fixing an issue with the original decompilation. In this specific situation, both things happen to match because of struct padding. Ideally we would have caught something like this in the decompilation phase many years ago, but it slipped through the cracks

Reply:
yeah, exactly. and now i know it is possible 🌞